doc/fluid/design/dist_train/distributed_lookup_table_design.md

```diff
@@ -44,111 +44,39 @@ $$W = f(W, W')$$
-optimization algorithm $f$ runs on the storage service.
-
-- Pro: parameter servers do not retrieve W(x) from the storage
-  service, thus saves half network communication.
-- Con: the storage service needs to be able to run the optimization
-  algorithm.
-
-## Distributed Sparse Table in Fluid
-
-For another design, we can implement a distributed sparse table in Fluid,
-and don't need to maintain an external storage component while training.
-
-You may need to read Fluid [Distributed Training Architecture](./distributed_architecture.md)
-and [Parameter Server](./parameter_server.md) before going on.
-
-![fluid lookup remote table](./src/fluid_lookup_remote_table.png)
-
-Partition a large table into multiple pserver instances
-1. `DistributeTranspiler` would split the table partitioned into some small
-table blocks with some partitioned algorithms such as
-[RoundRobin](https://en.wikipedia.org/wiki/Round-robin_scheduling),
-[Hash](https://en.wikipedia.org/wiki/Hash) and etc...
-1. For some cases, the range of input `Ids` is very wide and unpredictable, so the sparse
-table would be able to fill a new value for the id that didn't appear before with
-zero, uniform random or Gaussian distribution.
-
-For each Trainer's training process:
-1. In the forward pass, we use `pre-fetch` op to pre-fetch parameter blocks according to the
-input `Ids` from PServers instead of the local `lookup_table` op, and then merge the blocks
-into a parameter `W`.
-1. Compute `GRAD@W'` in the backward pass using the pre-fetched `W` and send it to PServer to
-execute the optimize pass.
-
-## Conclusion
-
-Let us do the "storage service does not optimize" solution first, as a
-baseline at least, because it is easier to use a well-optimized
-distributed storage service like memcached.  We can do the "storage
-service does optimize" solution later or at the same time, which, if
-implemented carefully, should have better performance than the former.
+## Distributed Lookup Table
+### Problem 1: The lookup table may be very large.
+
+ In the condition like the search engine and recommendation system, the number of feature ID may be very large, see 1000000000, then for a lookup table of size 8, the total size of the table is:
+
+ ```
+ 100,000,000,000 * 8 * 4.0 = 2980.23 GB
+ ```
+
+### Solution: Distributed storage
+
+1. Paddle use SelectedRows as the storage format for the lookup table, the lookup table parameter will be split to multi-machine according to the hash of the feature ID, and data will also be split and send to the same machine to prefetch the parameter.
+
+1. For common parameters, the trainer will get the whole parameter for training, but for the big lookup table, the trainer can not store the whole parameter, but the input data feature is very sparse, so every time we only need a few parameters for training, so we use `prefetch_op` to only prefetch the parameter needed to trainer.
+
+### Problem 2. The Id in the lookup table is not sure before training.
```

Comment:
Another problem: the model save cannot be done on a single machine.

Reply:
done